core/textile/sync/restore.go

```diff
@@ -0,0 +1,48 @@
+package sync
+
+import (
+	"context"
+
+	"github.com/FleekHQ/space-daemon/core/events"
+	"github.com/FleekHQ/space-daemon/log"
+)
+
+// replicate a local thread on the hub
+func (s *synchronizer) restoreBucket(ctx context.Context, bucketSlug string) error {
+
+	bucket, err := s.getBucket(ctx, bucketSlug)
+	if err != nil {
+		log.Error("Error in ListDir", err)
+		return err
+	}
+
+	dir, err := bucket.ListDirectory(ctx, "")
+	if err != nil {
+		log.Error("Error in ListDir", err)
+		return err
+	}
+
+	dirPaths := make([]string, 0)
+	for _, item := range dir.Item.Items {
+		dirPaths = append(dirPaths, item.Path)
+	}
+
+	mirrorFiles, err := s.model.FindMirrorFileByPaths(ctx, dirPaths)
+	if err != nil {
+		log.Error("Error fetching mirror files", err)
+		return err
+	}
+
+	for _, m := range mirrorFiles {
+		if err = s.downloadFileFromRemote(ctx, m.BucketSlug, m.Path); err != nil {
```

Comment:
Maybe an option we can do is span a "DownloadFile" task for each file instead of doing it all in one task

Reply:
@dmerrill6 

Done, as per our sync in using `restoreFileTask` etc.